classes/checkout/PaymentOptionsFinder.php

```diff
@@ -31,43 +31,31 @@
 
 class PaymentOptionsFinderCore extends HookFinder
 {
+    /**
+     * Collects available payment options from three different hooks
+     * 
+     * @return array An array of available payment options
+     * 
+     * @see HookFinder::find()
+     */
     public function find() //getPaymentOptions()
     {
         // Payment options coming from intermediate, deprecated version of the Advanced API
         $this->hookName = 'displayPaymentEU';
         $rawDisplayPaymentEUOptions = parent::find();
-
-        if (!is_array($rawDisplayPaymentEUOptions)) {
-            $rawDisplayPaymentEUOptions = array();
-        }
-
-        $displayPaymentEUOptions = array_map(
+        $paymentOptions = array_map(
             array('PrestaShop\PrestaShop\Core\Payment\PaymentOption', 'convertLegacyOption'),
             $rawDisplayPaymentEUOptions
         );
 
-        // Payment options coming from regular Advanced API
+        // Advanced payment options coming from regular Advanced API
         $this->hookName = 'advancedPaymentOptions';
-        $advancedPaymentOptions = parent::find();
-        if (!is_array($advancedPaymentOptions)) {
-            $advancedPaymentOptions = array();
-        }
+        $paymentOptions = array_merge($paymentOptions, parent::find());
 
         // Payment options coming from regular Advanced API
         $this->hookName = 'paymentOptions';
         $this->expectedInstanceClasses = array('PrestaShop\PrestaShop\Core\Payment\PaymentOption');
-        $newOption = parent::find();
-        if (!is_array($newOption)) {
-            $newOption = array();
-        }
-
-        $paymentOptions = array_merge($displayPaymentEUOptions, $advancedPaymentOptions, $newOption);
-
-        foreach ($paymentOptions as $paymentOptionKey => $paymentOption) {
-            if (!is_array($paymentOption)) {
-                unset($paymentOptions[$paymentOptionKey]);
-            }
-        }
```

Comment:
Thanks for the refacto, however I wonder if you don't remove a safety check on these 5 lines.

This needs to be double checked.

Reply:
As far as I can see parent::find() does all the checks, so I removed doubled safety checks hopefully :D

Reply:
Sorry I had time to review it. You where absolutely right of course. Thanks for the review, I made another commit